### PR TITLE
Native Transfer Policy Test

### PR DESCRIPTION
**Pull request by remedcu**
## TLDR
Added Native Transfer Policy Test

## LLM Description
This pull request introduces a comprehensive test suite for the `NativeTransferPolicy` contract, focusing on its integration with the `SafePolicyGuard` and ensuring proper policy configuration. The changes include new test cases to validate allowed and disallowed operations, as well as edge cases for policy configuration.

### Integration with `SafePolicyGuard`:

* Added tests to verify that native transfers with a value greater than zero are allowed, ensuring correct policy configuration and execution.
* Implemented tests to confirm that native transfers with a value of zero are rejected, enforcing the policy's constraints.
* Included tests to prevent non-native transfer transactions (e.g., transactions with data payloads), ensuring compliance with the native transfer policy.

### Policy Configuration:

* Added a test to ensure that only native transfer transactions can be configured, rejecting configurations with non-zero selectors.
* Implemented a test to restrict policy configurations to `CALL` operations, disallowing unsupported operations like `DELEGATECALL`.

### CODE REVIEW

File: test/nativeTransferPolicy.spec.ts [hunk 1]
```diff
@@ -0,0 +1,216 @@
...
+      })
+
+      // Enable the guard on safe
+      await execTransaction({
+        owners: [owner],
+        safe,
+        to: await safe.getAddress(),
+        data: safe.interface.encodeFunctionData('setGuard', [await safePolicyGuard.getAddress()])
+      })
+
```
[the remark this answers is not in the document]
Comment (remedcu) [reply test/nativeTransferPolicy.spec.ts:73]: Good Idea, maybe in a different PR.

File: test/nativeTransferPolicy.spec.ts [hunk 2]
```diff
@@ -0,0 +1,216 @@
...
+        to: await safePolicyGuard.getAddress(),
+        data: safePolicyGuard.interface.encodeFunctionData('configureImmediately', [configurations])
+      })
+
+      // Enable the guard on safe
+      await execTransaction({
+        owners: [owner],
+        safe,
+        to: await safe.getAddress(),
+        data: safe.interface.encodeFunctionData('setGuard', [await safePolicyGuard.getAddress()])
```
[the remark this answers is not in the document]
Comment (remedcu) [reply test/nativeTransferPolicy.spec.ts:71]: Won't implement here.
